### PR TITLE
fix: memory leak on Layer_Freetype

### PR DESCRIPTION
**Pull request by rodolforg**

### CODE REVIEW

File: synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp [hunk 1]
```diff
@@ -264,6 +252,99 @@ class FaceCache {
...
+	}
+
+private:
+	static void
+	self_destroy(void* object)
+	{
+		FT_Face face = static_cast<FT_Face>(object);
+		FaceMetaData* meta_data = static_cast<FaceMetaData*>(face->generic.data);
+		hb_font_destroy(meta_data->font);
+		delete meta_data;
```
Comment (ice0) [inline synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp:337]: Don't you need to clear `face->generic.data` here?
I mean:
```
face->generic.data = nullptr;
```

Comment (rodolforg) [reply synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp:338]: I'll try it

Comment (rodolforg) [reply synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp:338]: It seems it worked :D rebasing it

**Comment by rodolforg** [comment]
(force pushed to remove unused lines)

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**